### PR TITLE
Added to the commentary section

### PR DESCRIPTION
**Pull request by dotemacs**
I found out about your package via this MELPA PR: https://github.com/melpa/melpa/pull/4168

It looks super useful and I'd like to see it added to MELPA :)

So in order to help it make it happen, I'm creating this PR below.

Added a table demonstrating the usage of the package.

Opted for the table because I figured that it would make the advantage
of the package obvious to the user.

The above is to settle the first bullet point: https://github.com/melpa/melpa/pull/4168#issuecomment-241358267

> In the Commentary section, please don't just refer people to the github page. There should be some text in this section to describe what the package is and how to use it, so maybe copy some of the README into the Commentary.

### CODE REVIEW

**Comment by dotemacs** [comment]
As for the second point and comment added by @tarsius:

> But you should add a comment which states when you last synchronized your definition with the original and what Emacs version that corresponds to.

Could you add something like:

``` emacs-lisp
Package-Requires: ((emacs "24.5"))
```

if that was the version of Emacs you've tested it on, to the header: https://github.com/SanCoder-Q/y-ret-or-n-p.el/blob/master/y-ret-or-n-p.el#L8.

And then also add a comment here: https://github.com/SanCoder-Q/y-ret-or-n-p.el/blob/master/y-ret-or-n-p.el#L30

If you're too busy, and you'd like me to add those changes via a further commit to this PR, please let me know and I'd gladly help.

Thank you

File: y-ret-or-n-p.el [hunk 1]
```diff
@@ -23,7 +23,12 @@
 ;; Public License.
 
 ;;; Commentary:
-;; See readme (http://github.com/sancoder-q/y-ret-or-n-p.el) for installation and usage
+;;
+;; | Command       | You have to type                        |
+;; |---------------+-----------------------------------------|
+;; | yes-or-no-p   | 'yes' or 'no', followed by RET          |
+;; | y-or-no-p     | 'y' or 'n', followed by RET             |
+;; | y-ret-or-no-p | RET to confirm straight away or 'n' RET |
```
Comment (tarsius) [inline y-ret-or-n-p.el]: The package does not even define a function by that name. And I very much hope the function it does define does not behave the way you describe it. Making it _harder_ to cancel than to confirm would be strange, after all canceling by mistake is usually less dangerous than confirming by mistake.

Comment (dotemacs) [reply y-ret-or-n-p.el]: > The package does not even define a function by that name.

Well spotted, fixed now.

``` emacs-lisp
;; | Command       | You have to type                        |
;; |---------------+-----------------------------------------|
;; | yes-or-no-p   | 'yes' or 'no', followed by RET          |
;; | y-or-n-p      | 'y' or 'n', followed by RET             |
;; | y-ret-or-n-p  | RET to confirm straight away or 'n' RET |
```

**Comment by SanCoder-Q** [comment]
@dotemacs Thank you for your contribution. It's a great idea to add a description table to illustrate how the package works. I will merge it and update my PR https://github.com/melpa/melpa/pull/4168

@tarsius First, I agree with you. It indeed increases the risk of confirming by mistake. However, I think it should be fine for two main reasons. One is that I didn't remove the `yes-or-no-p` and `y-or-n-p`, so anyone who is using this package should know it and take the risk. I think people should have a choice to decide how much effort it should take when they want to do something that causes an inquiry. The other reason is from my point of view, most people around me who use emacs for a while feel that press two keys to confirm is always painful. And this is why I sent the PR in the first place. But I still think your notice is important, so I will add a warning to alert people who want to use it should know the risk you mentioned.

**Comment by tarsius** [comment]
I have looked at this again and noticed that what your package does can essentially be achieved  using just

```lisp
(define-key query-replace-map "\r" 'act)
```

I don't think it is necessary to mention "RET" in the prompt because, beside "y", there already are a few other keys that can be used, namely "SPC" and "Y", and they also aren't mentioned in the prompt.

**Comment by tarsius** [comment]
@SanCoder-Q You misunderstood me. What I was trying to say is that the description "RET to confirm straight away or 'n' RET"  does not make any sense, or that if it does accurately describe what `y-ret-or-n-p` does, then the name of that function would be misleading and behavior rather wacky.

Luckily it's only the description that is wrong, the implementation does the reasonable thing. So an accurate description would be: "To confirm press "y", "Y", "SPC", or (that's what this function offers over `y-or-n-p`) "RET". To abort press or "n", "N", or "DEL".

The description given by @dotemacs sais that to abort you have to press "n" _followed by_ "RET". That's what I was trying to point out when I said

> Making it **harder** to cancel than to confirm would be strange, [because this package is about making it **easier to confirm** - why would you _at the same time additionally_ make it **harder to abort**? Shouldn't aborting be kept just as easy/hard as before?]

**Comment by SanCoder-Q** [comment]
@tarsius Yes, you are absolutely right. My bad, I will update the description right way.

**Pull request merged**